### ISSUE TITLE
AjaxSpider with PhantomJS fails to start

### ISSUE THREAD

**Issue by kurobeats** [zaproxy/zaproxy#2636]
I think my terminal output will detail it best. Just trying to spider the google home page:

```
59764 [Thread-21] INFO org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Running crawljax targeting https://www.google.com
59767 [Thread-21] INFO org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Starting proxy...
59780 [Thread-21] INFO org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Proxy started, listening at port [39003].
60064 [Thread-21] INFO com.crawljax.core.plugin.Plugins  - Loaded org.zaproxy.zap.extension.spiderAjax.SpiderThread$DummyPlugin@7d261d30 as a OnBrowserCreatedPlugin
60127 [Thread-21] INFO org.openqa.selenium.phantomjs.PhantomJSDriverService  - executable: /usr/bin/phantomjs
60127 [Thread-21] INFO org.openqa.selenium.phantomjs.PhantomJSDriverService  - port: 8570
60127 [Thread-21] INFO org.openqa.selenium.phantomjs.PhantomJSDriverService  - arguments: [--proxy-type=http, --proxy=127.0.0.1:39003, --ssl-protocol=any, --ignore-ssl-errors=yes, --webdriver-logfile=/home/USER/.ZAP//phantomjsdriver.log, --webdriver-loglevel=WARN, --webdriver=8570, --webdriver-logfile=/usr/share/zaproxy/phantomjsdriver.log]
60127 [Thread-21] INFO org.openqa.selenium.phantomjs.PhantomJSDriverService  - environment: {}
Unable to open file '/usr/share/zaproxy/phantomjsdriver.log'

  phantomjs://platform/fs.js:79 in open
  phantomjs://platform/logger.js:95 in addLogFile
  phantomjs://platform/config.js:76 in apply
  phantomjs://platform/config.js:98 in init
80308 [Thread-21] WARN org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Failed to start browser phantomjs
com.google.inject.ProvisionException: Guice provision errors:

1) Error in custom provider, org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: 'sylar', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.6.3-1-ARCH', java.version: '1.8.0_92'
Driver info: driver.version: PhantomJSDriver
  at com.crawljax.di.ConfigurationModule.configure(ConfigurationModule.java:47)
  while locating com.crawljax.browser.EmbeddedBrowser
    for parameter 0 at com.crawljax.core.CrawlerContext.<init>(CrawlerContext.java:32)
  while locating com.crawljax.core.CrawlerContext
    for parameter 0 at com.crawljax.core.Crawler.<init>(Crawler.java:73)
  while locating com.crawljax.core.Crawler
    for parameter 2 at com.crawljax.core.CrawlTaskConsumer.<init>(CrawlTaskConsumer.java:30)
  while locating com.crawljax.core.CrawlTaskConsumer

1 error
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:987)
	at com.crawljax.core.CrawlController.call(CrawlController.java:65)
	at com.crawljax.core.CrawljaxRunner.call(CrawljaxRunner.java:37)
	at org.zaproxy.zap.extension.spiderAjax.SpiderThread.run(SpiderThread.java:196)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.openqa.selenium.remote.UnreachableBrowserException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: 'sylar', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.6.3-1-ARCH', java.version: '1.8.0_92'
Driver info: driver.version: PhantomJSDriver
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:665)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:249)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:144)
	at org.openqa.selenium.phantomjs.PhantomJSDriver.<init>(PhantomJSDriver.java:115)
	at org.openqa.selenium.phantomjs.PhantomJSDriver.<init>(PhantomJSDriver.java:104)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getWebDriverImpl(ExtensionSelenium.java:271)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getWebDriver(ExtensionSelenium.java:224)
	at org.zaproxy.zap.extension.spiderAjax.SpiderThread$AjaxSpiderBrowserBuilder.get(SpiderThread.java:358)
	at org.zaproxy.zap.extension.spiderAjax.SpiderThread$AjaxSpiderBrowserBuilder.get(SpiderThread.java:329)
	at com.google.inject.util.Providers$3.get(Providers.java:109)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
	at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:84)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
	at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
	at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:84)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
	at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
	at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:84)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
	at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
	... 4 more
Caused by: java.lang.IllegalStateException: The process has not exited yet therefore no result is available ...
	at org.apache.commons.exec.DefaultExecuteResultHandler.getException(DefaultExecuteResultHandler.java:76)
	at org.openqa.selenium.os.UnixProcess.checkForError(UnixProcess.java:171)
	at org.openqa.selenium.os.CommandLine.checkForError(CommandLine.java:185)
	at org.openqa.selenium.remote.service.DriverService.waitUntilAvailable(DriverService.java:177)
	at org.openqa.selenium.remote.service.DriverService.start(DriverService.java:166)
	at org.openqa.selenium.phantomjs.PhantomJSCommandExecutor.execute(PhantomJSCommandExecutor.java:78)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:644)
	... 30 more
82526 [Thread-21] INFO org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Stopping proxy...
82627 [Thread-21] INFO org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Proxy stopped.
82628 [Thread-21] INFO org.zaproxy.zap.extension.spiderAjax.SpiderThread  - Finished crawljax targeting https://www.google.com
```

**Comment by thc202** [comment]
Which version of ZAP and Selenium add-on are you using?

**Comment by kurobeats** [comment]
ZAP is 2.5.0
Selenium is 5

**Comment by thc202** [comment]
OK, thanks.

**Comment by thc202** [comment]
Related issue detro/ghostdriver#383 (fixed in detro/ghostdriver#438).

**Comment by thc202** [comment]
Fixed included in version 6 of Selenium add-on.
